scripts/run-kani.sh

```diff
@@ -0,0 +1,206 @@
+    echo "  --kani-args  <command arguments to kani>  Optional: Arguments to pass to the command. Simply pass them in the same way you would to the Kani binary. This should be the last argument."
+    exit 1
+}
+
+# Initialize variables
+command_args=""
+path=""
+
+# Parse command line arguments
+# TODO: Improve parsing with getopts
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        -h|--help)
+            usage
+            ;;
+        -p|--path)
+            if [[ -n $2 ]]; then
+                path=$2
+                shift 2
+            else
+                echo "Error: Path argument is missing"
+                usage
+            fi
+            ;;
+        --kani-args)
+            shift
+            command_args="$@"
+            break
+            ;;
+        *)
+            break
+            ;;
+    esac
+done
+
+# Set working directory
+if [[ -n "$path" ]]; then
+    if [[ ! -d "$path" ]]; then
+        echo "Error: Specified directory does not exist."
+        usage
+        exit 1
+    fi
+    WORK_DIR=$(realpath "$path")
+else
+    WORK_DIR=$(pwd)
+fi
+
+cd "$WORK_DIR" || exit 1
+
+# Default values
+DEFAULT_TOML_FILE="tool_config/kani-version.toml"
+DEFAULT_REPO_URL="https://github.com/model-checking/kani.git"
+DEFAULT_BRANCH_NAME="features/verify-rust-std"
+
+# Use environment variables if set, otherwise use defaults
+TOML_FILE=${KANI_TOML_FILE:-$DEFAULT_TOML_FILE}
+REPO_URL=${KANI_REPO_URL:-$DEFAULT_REPO_URL}
+BRANCH_NAME=${KANI_BRANCH_NAME:-$DEFAULT_BRANCH_NAME}
+
+os_name=$(uname -s)
+
+# Function to read commit ID from TOML file
+read_commit_from_toml() {
+    local file="$1"
+    if [ ! -f "$file" ]; then
+        echo "Error: TOML file not found: $file" >&2
+        exit 1
+    fi
+    local commit=$(grep '^commit *=' "$file" | sed 's/^commit *= *"\(.*\)"/\1/')
+    if [ -z "$commit" ]; then
+        echo "Error: 'commit' field not found in TOML file" >&2
+        exit 1
+    fi
+    echo "$commit"
+}
+
+clone_repo() {
+    local repo_url="$1"
+    local directory="$2"
+    local branch="$3"
+    local commit="$4"
+    git clone --depth 1 -b "$branch" "$repo_url" "$directory"
+    cd "$directory"
+    git checkout "$commit"
+    cd - > /dev/null
+}
+
+checkout_commit() {
+    local directory="$1"
+    local commit="$2"
+    cd "$directory"
+    git checkout "$commit"
+    cd ..
+}
+
+get_current_commit() {
+    local directory="$1"
+    if [ -d "$directory/.git" ]; then
+        git -C "$directory" rev-parse HEAD
+    else
+        echo ""
+    fi
+}
+
+build_project() {
+    local directory="$1"
+    cd "$directory"
+
+    if [ "$os_name" == "Linux" ]; then
+        ./scripts/setup/ubuntu/install_deps.sh
+    elif [ "$os_name" == "Darwin" ]; then
+        ./scripts/setup/macos/install_deps.sh
+    else
+        echo "Unknown operating system"
+    fi
+
+    cargo build-dev --release
+    cd ..
```

Comment:
Should likely be `cd -` (or use `pushd`/`popd` as suggested above) as otherwise you are now in some random place when `$directory` is not just the immediate subdirectory to the working directory of the caller.